versions/schematic-2.md

```diff
@@ -0,0 +1,235 @@
+
+## Introduction
+
+This specification defines a format which describes an area of a [Minecraft](http://minecraft.net) world for the purpose of serialization to disk and deserialization from disk. It is designed in order to allow maximum cross-compatibility between platforms, versions, and various states of modification.
+
+## Revision History
+
+Version | Date | Notes
+---|:---:|---
+1 | 2016-08-23 | Initial Version
+2 | 2018-08-19 | More features, easier handling, see [Changelog](#changelog)
+
+## Specification
+
+### Format
+
+All field names in the specification are **case sensitive**.
+
+#### <a name="formFile"></a>File
+
+The structure described by this specification is persisted to disk using the [Named Binary Tag](http://minecraft.gamepedia.com/NBT_format) (NBT) format (also see [here](https://wiki.vg/NBT)). Before writing to disk the NBT data must be compressed using the [GZip](https://www.gnu.org/software/gzip/) data compression algorithm. The highly recommended file extension for files using this specification is `.schem` or `.spongeschem` (chosen so as to not conflict with the legacy `.schematic` format allowing easy distinction between the two).
+
+#### <a name="formId"></a>Identification string
+
+Resources identified by a resource location have an identifier string which is made up of two sections, the domain and the name, written as `domain:name`. If the domain is not specified then it is assumed to be `minecraft`. For example `planks` would identify the same resource as `minecraft:planks` as the `minecraft:` prefix is implicit.
+
+### Schema
+
+#### <a name="schematicObject"></a>Schematic Object
+
+This is the nameless root object of the schematic. It shall be a [NBT](#formFile) tag compound.
+
+##### Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="schematicVersion"></a>Version | `NBT int` | **Required.** Specified the format version being used. It may be used to provide validation and auto-conversion from older versions. The current version is `2`. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicContentVersion"></a>ContentVersion | `NBT int` | **Required.** A version identifier for the contents. Representing the minecraft version, when serialized. Used for providing backwards compatibility. [Values are specified by Mojang](https://minecraft-de.gamepedia.com/Versionen#Version-IDs). Mojangs data converter for older versions will need this value for proper conversion. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicMetadata"></a>Metadata | [Metadata Object](#metadataObject) as `NBT compound` | **Optional.** Provides metadata about the schematic. This shall be saved as [NBT](#formFile) tag compound.
+<a name="schematicWidth"></a>Width | `NBT int` | **Required.** Specifies the width (the amount of blocks along the X-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
```

Comment:
4294967295 - 65535 = 4294901760
Why loose over 4 billion blocks if we could support it without any problem?
I see no advantage in using Short instead of Integer.
Also keep in mind, it restricts not only a whole cuboid (65535 x 65535 x 65535), but also in one dimension, too (65535 x 1 x 1).
Possible performance issues of any implementation trying to handle this much blocks are not our problem. Any implementation must individually care about how to handle (or not to handle / block) this much blocks.
Maybe we should add somewhere, that handling of a huge amount of blocks (more than x blocks) MUST be done within an async thread. Since minecraft implementations can only use the main sync thread to modify the world, there MUST be a parameter like X blocks per tick.